src/Combinatorics/Graphs/functions.jl

```diff
@@ -79,68 +79,65 @@ function graph_from_adjacency_matrix(::Type{Undirected}, G::Union{MatElem, Matri
   return g
 end
 
-function graph_from_adjacency_matrix(::Type{Directed}, G::Union{MatElem, Matrix})
-  n = nrows(G)
-  @req nrows(G)==ncols(G) "not a square matrix"
-  g = Graph{Directed}(n)
-  for i in 1:n
-    for j in 1:n
-      if isone(G[i,j])
-        add_edge!(g, i, j)
-      else
-        iszero(G[i,j]) || error("not an adjacency matrix")
-      end
-    end
-  end
-  return g
-end
 
 _has_node(G::Graph, node::Int64) = 0 < node <= nv(G)
 
 @doc raw"""
     add_edge!(g::Graph{T}, s::Int64, t::Int64) where {T <: Union{Directed, Undirected}}
 
 Add edge `(s,t)` to the graph `g`.
+Return `true` if operation was successful, `false` otherwise.
```

Comment:
What does it mean that it was "not successful"? That the edge was already there? Or can it fail for other reasons? Which?

If the former, then I would suggest to make that explicit, e.g.:
```suggestion
Return `true` if there was no edge there before, `false` otherwise.
```